include/utils/OpenPolylineStitcher.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2024 Ultimaker B.V.
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef UTILS_OPEN_POLYLINE_STITCHER_H
+#define UTILS_OPEN_POLYLINE_STITCHER_H
+
+#include "PolylineStitcher.h"
+#include "geometry/open_lines_set.h"
+#include "geometry/shape.h"
+
+namespace cura
+{
+
+using OpenPolylineStitcher = PolylineStitcher<OpenLinesSet, Shape, OpenPolyline, Point2LL>;
```

Comment:
It's a good C++ practice to have a single type declaration inside a single header file. This avoids exponential header inclusion that makes builds longer and can lead to circular inclusion. Moreover, if you want to change this type to e.g. an actual class, you can do it safely inside this file.

Reply:
> I know you might be OK with different classes in different files, since you have that in different parts in this PR.

In fact, I am not. I kept some of them because it is legacy code, but I take it as a personal rule to always have one header by type (or maybe sometimes for basic typedefs, but even for that I tend to do it less and less). That makes projects with 100s of header files, but they are small, easy to read and don't include lot of other files. And when you want to use a type, you exactly know where to find it.

Reply:
I was just confused by what seemed like an inconsistency.